src/services/textChanges.ts

```diff
@@ -83,14 +83,18 @@ namespace ts.textChanges {
         delta?: number;
     }
 
+    export interface ChangeNodesOptions {
+        nodesSeparator?: string;
+    }
+
     export type ChangeNodeOptions = ConfigurableStartEnd & InsertNodeOptions;
 
     interface Change {
         readonly sourceFile: SourceFile;
         readonly range: TextRange;
         readonly useIndentationFromFile?: boolean;
-        readonly node?: Node;
-        readonly options?: ChangeNodeOptions;
+        readonly node?: Node | Node[];

```

Comment:
Changed this to be more explicit, but we actually do allocate a fair number of these and I think it's worth avoiding the array allocation in the (much more common) 1-element case.